demo/src/lt.tooltip.js

```diff
@@ -17,18 +23,55 @@ angular.module('lt.tooltip', [])
             $tooltipEl = angular.element(el.querySelector('.tooltip')),
             $transcluded = angular.element(el.querySelector('.transcluded')),
             rect = el.getBoundingClientRect();
-
+        
         return {
             top: rect.top,
             left: rect.left,
             tooltipHeight: $tooltipEl[0].offsetHeight,
             tooltipWidth: $tooltipEl[0].offsetWidth,
             transcludedHeight: $transcluded[0].offsetHeight,
-            transcludedWidth: $transcluded[0].offsetWidth
+            transcludedWidth: $transcluded[0].offsetWidth,
+            windowHeight: $window.innerHeight,
+            windowWidth: $window.innerWidth
         };
     };
 
-    // Calculate the position of the tooltip. Returns the position as an object.
+    // Determine from the given position and dimensions if the tooltip will
+    // end up completely within the window (and be visible).
+    // Returns true if the tooltip will be visible, false otherwise.
+    var isWithinVisibleArea = function (position, dimensions) {
+        var top = position.top;
+        var right = position.left + dimensions.tooltipWidth;
+        var bottom = position.top + dimensions.tooltipHeight;
+        var left = position.left;
+        return top >= 0 && left >= 0 &&
+            right < $window.innerWidth && bottom < $window.innerHeight;
+    };
+
+    // Removes a given direction from the given array of directions.
+    // Returns the updated array.
+    var removeDirection = function (direction, directions) {
+        return directions.map(function(dir) {
+            if (dir === direction) {
+                return;
+            }
+            return dir;
+        });
+    };
+
+    // Retrieve the next direction in the priority list, as a string.
+    var nextDirection = function (direction, directionPriorities) {
+        var directions = directionPriorities || DEFAULT_DIRECTION_PRIORITY;
+        var index = directions.indexOf(direction);
+        ++index;
+        if (index >= directions.length) {
+            index = 0;
+        }
+        return directions[index];
+    };
```

Comment:
I see you programmed with the assumption that directionPriorities are going to be modified by the user. Nice.